### ISSUE TITLE
Rename transaction

### ISSUE THREAD

**Issue by baudouxbenjamin** [open-telemetry/opentelemetry-android#325]
Hello, 

We're using elastic on Android and for now every transactions are under something like `GET`, `POST`, etc.
I'd like to rename my transactions to `GET my-service.be/data` for example to make it easier to find.

Is it possible? If not, is it planned in a future release?

Thanks!

**Comment by breedx-splk** [comment]
Hi @baudouxbenjamin . The names of the spans generated by this instrumentation should comply with the [OpenTelemetry HTTP semantic conventions](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name). 

If you check that link, you'll see that the guidance for client http spans is:

> HTTP client spans have no http.route attribute since client-side instrumentation is not generally aware of the “route”, and therefore HTTP client span names SHOULD be [{method}](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#method-placeholder).

So I don't think we have any real intention to change this. You can always put manual instrumentation around these call sites, if you'd like to get more details in the name.

**Comment by LikeTheSalad** [comment]
I agree we should stick to the semconv, however, it does mention that we could add the HTTP route (if we knew what it was) though we don't know, so it makes sense we don't do it and I don't think OTel Android should try to make any "smart" decisions on what the HTTP route should look like. 

Still, I think we could at least provide some APIs to allow users to add the route themselves, maybe in a similar fashion to what someone's [suggested doing](https://github.com/open-telemetry/opentelemetry-android/issues/211#issuecomment-2045576938) for screen names.

**Comment by LikeTheSalad** [comment]
There has been some progress around this issue in the Java SIG and also [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11355). I'm planning to add a config option for it soon.

**Comment by breedx-splk** [comment]
@LikeTheSalad do you think we can close this now?

**Comment by LikeTheSalad** [comment]
> [@LikeTheSalad](https://github.com/LikeTheSalad) do you think we can close this now?

Hmm. I guess it depends on what's needed. The config option is there now in the upstream instrumentation repo, so maybe we can create a wrapper for it here, similarly to what was done for attributes on [this PR](https://github.com/open-telemetry/opentelemetry-android/pull/867). However, that would only solve the problem for the HTTP spans generated via those automatic instrumentations.

For a global solution, I think the current best option would be to modify HTTP span names in a custom SpanProcessor. I think this option is already available via our existing APIs, so if that's what's needed, then we can close this issue. Otherwise, if we wanted to create configs for each HTTP auto-instrumentation impl, then it'll have to wait until that's added. I think the best option is the former as it's a global solution, so I'm up for closing this issue unless someone would like to provide a reason why a config per auto-instrumentation would be preferred.
